src/CMakeLists.txt

```diff
@@ -35,15 +35,20 @@ set(SOURCE_FILES
   ${CMAKE_CURRENT_SOURCE_DIR}/version.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/model.cpp
   ${CMAKE_CURRENT_SOURCE_DIR}/component.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/componententity.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/namedentity.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/entity.cpp
```

Comment:
I would favour listing files in alphabetical order. It makes it easier to check/review/etc.